### DISCUSSION TITLE
HotReload is ok but not enough.. Please, Bring back LiveReload or similar

### DISCUSSION THREAD

**Discussion by EmilAlipiev** [xamarin/Xamarin.Forms#11451] [General]
Hot reload is only partly ok. mostly doesn't work on popups, masterdetail, tabs, titleview, collectionview etc etc. 
Most importantly it doesn't support UWP. It has been such a long time since you promised and delayed Hotreload for UWP but i feel that it will never happen before Maui.
**And the biggest drawback of Hotreload over older Livereload** is that it only works in debug mode. Livereload supported working without debugging. I was able to make a change and see my changes on multiple emulators simultaneously. Such an amazing feature indeed. It trashes out all those flutter, react native hotreloads. Why would you remove it?
At least give use livereload as optional extension as open-source so people can decide which one to use? I really miss this feature to be able to test my ui changes on multiple emulators.

**Comment by davidortinau** [comment]
There are currently no plans to do hot reloading without the debugger. This could of course change in the future, but we have found this to be MUCH more reliable and work across a broader spectrum of corporate environments.

Xamarin.Forms XAML Hot Reload for UWP will be coming very soon, in the next few releases of VS. I heard it was accidentally omitted from the [roadmap](https://docs.microsoft.com/en-us/visualstudio/productinfo/vs-roadmap), but should be there.

XAML Hot Reload supports [multiple devices today](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/xaml/hot-reload#reload-on-multiple-platforms-at-once).

If you liked how Live Reload worked, check out [Gorilla Player](https://grialkit.com/gorilla-player/), [Hot Reload](https://github.com/AndreiMisiukevich/HotReload), or [LiveSharp](https://www.livesharp.net/).

Please be sure to file issues specific to hot reloading different controls from Visual Studio. Those ought to all work.

**Reply by EmilAlipiev** [reply]
thanks for the uwp update. seriously looking forward to it.

**Comment by EmilAlipiev** [comment]
Latest update VS 201916.7.0 came with screenshot below on my xamarin.forms project for uwp. is it mistake? it doesn't work for me :( 

![image](https://user-images.githubusercontent.com/10422347/89526482-1c92f900-d7e8-11ea-8e84-77298385cd86.png)

**Reply by MitchBomcanhao** [reply]
@EmilAlipiev , hot reload has always shown when debugging UWP apps - its likely just a side effect of using the native UWP debugging tools which have supported hot reload for several years now. It doesn't mean that it works for Xamarin forms apps.
